api/src/rateLimiters.js

```diff
@@ -0,0 +1,39 @@
+class RateLimiter {
+  constructor(limitPerSecond) {
+    this.limit = limitPerSecond;
+    this.calls = 0;
+    this.queue = [];
+    setInterval(() => this._release(), 1000);
+  }
+
+  _release() {
+    this.calls = 0;
+    while (this.calls < this.limit && this.queue.length) {
+      console.log("🚀 ~ file: rateLimiters.js:12 ~ RateLimiter ~ _release ~ this.queue.length:", this.queue.length);
+      const nextCall = this.queue.shift();
+      nextCall();
+      this.calls += 1;
```

Comment:
I would suggest to swap those 2 lines, because the execution of the function could take a little while but you want to increment the calls right away to queue the newly incoming calls